app/views/invoices/_form.html.erb

```diff
@@ -27,10 +27,14 @@
 
   <tr>
     <td class="subheading"><%= f.label :status, "Status" %>:</td>
-    <% if can? :status, f.object %>
+    <% if current_member.is_at_least? :tracker_management %>
       <td><%= f.select :status, Invoice::Invoice_Status_Group_All %></td>
-    <% elsif f.object.new_record? %>
-      <td><%= f.select :status, Invoice::Invoice_Status_Group_Exec %></td>
+    <% elsif current_member.is_at_least? :exec%>
+      <% if f.object.status == "Received" || f.object.status == "Loan Agreement" %>
```

Comment:
Don't case on specific statuses here if you can use the `Invoice_Status_Group_All` and `Invoice_Status_Group_Exec` parts already defined.

This form filter is also not consistent with what you changed in `ability.rb`, which gives exec access to change all.